{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from tqdm import tqdm\n",
    "import os"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE37302"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse37302 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse37302/GSE37302.top.table.tsv\", sep=\"\\t\")\n",
    "gse37302 = gse37302[gse37302['adj.P.Val'] <= 0.05]\n",
    "gse37302_genes = list(set(gse37302['ORF'].values.tolist()))\n",
    "gse37302_genes =sorted([i for i in gse37302_genes if not \"nan\" in str(i).lower()])\n",
    "with open('gse37302_genes.txt','w') as handle:\n",
    "    for gen in gse37302_genes:\n",
    "        handle.write(gen + \"\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE39683"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>Description</th>\n",
       "      <th>converted_alias</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>namespace</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>933_at</td>\n",
       "      <td>1.870000e-45</td>\n",
       "      <td>9.510000e-50</td>\n",
       "      <td>-47.945383</td>\n",
       "      <td>99.964830</td>\n",
       "      <td>-7.208842</td>\n",
       "      <td>933</td>\n",
       "      <td>CD22 molecule</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2841_at</td>\n",
       "      <td>5.750000e-44</td>\n",
       "      <td>5.840000e-48</td>\n",
       "      <td>-44.697743</td>\n",
       "      <td>96.414938</td>\n",
       "      <td>-6.568676</td>\n",
       "      <td>2841</td>\n",
       "      <td>G protein-coupled receptor 18</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>941_at</td>\n",
       "      <td>2.670000e-37</td>\n",
       "      <td>4.070000e-41</td>\n",
       "      <td>-34.073867</td>\n",
       "      <td>82.246194</td>\n",
       "      <td>-5.895277</td>\n",
       "      <td>941</td>\n",
       "      <td>CD80 molecule</td>\n",
       "      <td>AC233723</td>\n",
       "      <td>PSMB6</td>\n",
       "      <td>proteasome 20S subunit beta 6 [Source:HGNC Sym...</td>\n",
       "      <td>AFFY_HC_G110,AFFY_HG_U95A,AFFY_HG_U95AV2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>941_at</td>\n",
       "      <td>2.670000e-37</td>\n",
       "      <td>4.070000e-41</td>\n",
       "      <td>-34.073867</td>\n",
       "      <td>82.246194</td>\n",
       "      <td>-5.895277</td>\n",
       "      <td>941</td>\n",
       "      <td>CD80 molecule</td>\n",
       "      <td>AK311768</td>\n",
       "      <td>PSMB6</td>\n",
       "      <td>proteasome 20S subunit beta 6 [Source:HGNC Sym...</td>\n",
       "      <td>AFFY_HC_G110,AFFY_HG_U95A,AFFY_HG_U95AV2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>941_at</td>\n",
       "      <td>2.670000e-37</td>\n",
       "      <td>4.070000e-41</td>\n",
       "      <td>-34.073867</td>\n",
       "      <td>82.246194</td>\n",
       "      <td>-5.895277</td>\n",
       "      <td>941</td>\n",
       "      <td>CD80 molecule</td>\n",
       "      <td>BC000835</td>\n",
       "      <td>PSMB6</td>\n",
       "      <td>proteasome 20S subunit beta 6 [Source:HGNC Sym...</td>\n",
       "      <td>AFFY_HC_G110,AFFY_HG_U95A,AFFY_HG_U95AV2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID     adj.P.Val       P.Value          t          B     logFC  \\\n",
       "0   933_at  1.870000e-45  9.510000e-50 -47.945383  99.964830 -7.208842   \n",
       "1  2841_at  5.750000e-44  5.840000e-48 -44.697743  96.414938 -6.568676   \n",
       "2   941_at  2.670000e-37  4.070000e-41 -34.073867  82.246194 -5.895277   \n",
       "3   941_at  2.670000e-37  4.070000e-41 -34.073867  82.246194 -5.895277   \n",
       "4   941_at  2.670000e-37  4.070000e-41 -34.073867  82.246194 -5.895277   \n",
       "\n",
       "   SPOT_ID                    Description converted_alias   name  \\\n",
       "0      933                  CD22 molecule            None   None   \n",
       "1     2841  G protein-coupled receptor 18            None   None   \n",
       "2      941                  CD80 molecule        AC233723  PSMB6   \n",
       "3      941                  CD80 molecule        AK311768  PSMB6   \n",
       "4      941                  CD80 molecule        BC000835  PSMB6   \n",
       "\n",
       "                                         description  \\\n",
       "0                                               None   \n",
       "1                                               None   \n",
       "2  proteasome 20S subunit beta 6 [Source:HGNC Sym...   \n",
       "3  proteasome 20S subunit beta 6 [Source:HGNC Sym...   \n",
       "4  proteasome 20S subunit beta 6 [Source:HGNC Sym...   \n",
       "\n",
       "                                  namespace  \n",
       "0                                       NaN  \n",
       "1                                       NaN  \n",
       "2  AFFY_HC_G110,AFFY_HG_U95A,AFFY_HG_U95AV2  \n",
       "3  AFFY_HC_G110,AFFY_HG_U95A,AFFY_HG_U95AV2  \n",
       "4  AFFY_HC_G110,AFFY_HG_U95A,AFFY_HG_U95AV2  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse39683_genecounts = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse39683/GSE39683.top.table.tsv\", sep=\"\\t\")\n",
    "gse39683_genecounts = gse39683_genecounts[gse39683_genecounts[\"adj.P.Val\"] <= 0.05]\n",
    "gene_probemapping = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse39683/gProfiler_hsapiens_28-04-2023_20-53-58.csv\")\n",
    "gene_probemapping.columns = [\"ID\"] + gene_probemapping.columns.tolist()[1:]\n",
    "gene_probemapping[\"ID\"] = [i.lower() for i in gene_probemapping[\"ID\"].values.tolist()]\n",
    "gse39683_genecounts1 = gse39683_genecounts.merge(gene_probemapping, on=[\"ID\"])\n",
    "gse39683_genecounts1_genes = sorted(list(set([i for i in gse39683_genecounts1[\"name\"] if not \"none\" in str(i).lower()])))\n",
    "with open(\"gse39683_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse39683_genecounts1_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "\n",
    "gse39683_genecounts1.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE46053"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse46053 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse46053/GSE46053.top.table.tsv\", sep=\"\\t\")\n",
    "gse46053 = gse46053[gse46053[\"adj.P.Val\"] <= 0.05]\n",
    "gse46053_genes = sorted(list(set([i for i in gse46053[\"ORF\"].values.tolist() if not \"none\" in i.lower()])))\n",
    "with open(\"gse46053_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse46053_genes:\n",
    "        handle.write(gen + \"\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE46609"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['BAG3', 'DNAJB1', 'HSPA1B', 'LCMT2']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse46609 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse46609/GSE46609.top.table.tsv\", sep=\"\\t\")\n",
    "gse46609 = gse46609[gse46609[\"adj.P.Val\"] <= 0.05]\n",
    "gse46609_genes = sorted(list(set([i for i in gse46609[\"Gene.symbol\"] if not \"nan\" in str(i).lower()])))\n",
    "gse46609_genes"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE72213"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>log2(fold change)</th>\n",
       "      <th>-log10(Pvalue)</th>\n",
       "      <th>converted_alias</th>\n",
       "      <th>name</th>\n",
       "      <th>description</th>\n",
       "      <th>namespace</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A_19_P00315529</td>\n",
       "      <td>-2.904</td>\n",
       "      <td>3.245</td>\n",
       "      <td>ENSG00000237943</td>\n",
       "      <td>PRKCQ-AS1</td>\n",
       "      <td>PRKCQ antisense RNA 1 [Source:HGNC Symbol;Acc:...</td>\n",
       "      <td>AGILENT_SUREPRINT_G3_GE_8X60K,AGILENT_SUREPRIN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A_19_P00315900</td>\n",
       "      <td>-2.383</td>\n",
       "      <td>3.029</td>\n",
       "      <td>ENSG00000234323</td>\n",
       "      <td>LINC01505</td>\n",
       "      <td>long intergenic non-protein coding RNA 1505 [S...</td>\n",
       "      <td>AGILENT_GPL19072,AGILENT_SUREPRINT_G3_GE_8X60K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A_19_P00317054</td>\n",
       "      <td>-3.146</td>\n",
       "      <td>3.748</td>\n",
       "      <td>ENSG00000228956</td>\n",
       "      <td>SATB1-AS1</td>\n",
       "      <td>SATB1 antisense RNA 1 [Source:HGNC Symbol;Acc:...</td>\n",
       "      <td>AGILENT_GPL19072,AGILENT_SUREPRINT_G3_GE_8X60K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A_19_P00317277</td>\n",
       "      <td>-3.194</td>\n",
       "      <td>3.274</td>\n",
       "      <td>ENSG00000281162</td>\n",
       "      <td>LINC01127</td>\n",
       "      <td>long intergenic non-protein coding RNA 1127 [S...</td>\n",
       "      <td>AGILENT_GPL19072,AGILENT_SUREPRINT_G3_GE_8X60K...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A_19_P00317639</td>\n",
       "      <td>-3.517</td>\n",
       "      <td>4.274</td>\n",
       "      <td>ENSG00000245164</td>\n",
       "      <td>LINC00861</td>\n",
       "      <td>long intergenic non-protein coding RNA 861 [So...</td>\n",
       "      <td>AGILENT_SUREPRINT_G3_GE_8X60K,AGILENT_SUREPRIN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               ID  log2(fold change)  -log10(Pvalue)  converted_alias  \\\n",
       "0  A_19_P00315529             -2.904           3.245  ENSG00000237943   \n",
       "1  A_19_P00315900             -2.383           3.029  ENSG00000234323   \n",
       "2  A_19_P00317054             -3.146           3.748  ENSG00000228956   \n",
       "3  A_19_P00317277             -3.194           3.274  ENSG00000281162   \n",
       "4  A_19_P00317639             -3.517           4.274  ENSG00000245164   \n",
       "\n",
       "        name                                        description  \\\n",
       "0  PRKCQ-AS1  PRKCQ antisense RNA 1 [Source:HGNC Symbol;Acc:...   \n",
       "1  LINC01505  long intergenic non-protein coding RNA 1505 [S...   \n",
       "2  SATB1-AS1  SATB1 antisense RNA 1 [Source:HGNC Symbol;Acc:...   \n",
       "3  LINC01127  long intergenic non-protein coding RNA 1127 [S...   \n",
       "4  LINC00861  long intergenic non-protein coding RNA 861 [So...   \n",
       "\n",
       "                                           namespace  \n",
       "0  AGILENT_SUREPRINT_G3_GE_8X60K,AGILENT_SUREPRIN...  \n",
       "1  AGILENT_GPL19072,AGILENT_SUREPRINT_G3_GE_8X60K...  \n",
       "2  AGILENT_GPL19072,AGILENT_SUREPRINT_G3_GE_8X60K...  \n",
       "3  AGILENT_GPL19072,AGILENT_SUREPRINT_G3_GE_8X60K...  \n",
       "4  AGILENT_SUREPRINT_G3_GE_8X60K,AGILENT_SUREPRIN...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse72213 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse72213/genes.tsv\", sep=\"\\t\")\n",
    "gene_probemapping = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse72213/gProfiler_hsapiens_28-04-2023_15-15-38.csv\")\n",
    "gene_probemapping.columns = [\"ID\"] + gene_probemapping.columns.tolist()[1:]\n",
    "gse72213 = gse72213.merge(gene_probemapping, on=[\"ID\"])\n",
    "gse72213_genes = sorted(list(set(gse72213[\"name\"].values.tolist())))\n",
    "with open(\"gse72213_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse72213_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "gse72213.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE78884"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>GB_LIST</th>\n",
       "      <th>GENE_NAME</th>\n",
       "      <th>SPOT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>467</td>\n",
       "      <td>0.000382</td>\n",
       "      <td>3.080000e-07</td>\n",
       "      <td>8.337315</td>\n",
       "      <td>6.9794</td>\n",
       "      <td>2.827543</td>\n",
       "      <td>NM_017957</td>\n",
       "      <td>EPN3</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>404</td>\n",
       "      <td>0.000382</td>\n",
       "      <td>5.940000e-07</td>\n",
       "      <td>-7.930107</td>\n",
       "      <td>6.3692</td>\n",
       "      <td>-1.999343</td>\n",
       "      <td>NM_021235</td>\n",
       "      <td>EPS15L1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>60</td>\n",
       "      <td>0.001135</td>\n",
       "      <td>2.640000e-06</td>\n",
       "      <td>-7.050291</td>\n",
       "      <td>4.9663</td>\n",
       "      <td>-2.089800</td>\n",
       "      <td>NM_005345, NM_005346</td>\n",
       "      <td>HSPA1A-HSPA1B</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>673</td>\n",
       "      <td>0.001750</td>\n",
       "      <td>7.150000e-06</td>\n",
       "      <td>-6.495496</td>\n",
       "      <td>4.0208</td>\n",
       "      <td>-2.664657</td>\n",
       "      <td>NM_016494</td>\n",
       "      <td>LOC51255</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>0.001750</td>\n",
       "      <td>8.120000e-06</td>\n",
       "      <td>6.426159</td>\n",
       "      <td>3.8993</td>\n",
       "      <td>1.346543</td>\n",
       "      <td>NM_178014</td>\n",
       "      <td>TUBB_HUMAN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ID  adj.P.Val       P.Value         t       B     logFC  \\\n",
       "0  467   0.000382  3.080000e-07  8.337315  6.9794  2.827543   \n",
       "1  404   0.000382  5.940000e-07 -7.930107  6.3692 -1.999343   \n",
       "2   60   0.001135  2.640000e-06 -7.050291  4.9663 -2.089800   \n",
       "3  673   0.001750  7.150000e-06 -6.495496  4.0208 -2.664657   \n",
       "4    6   0.001750  8.120000e-06  6.426159  3.8993  1.346543   \n",
       "\n",
       "                GB_LIST      GENE_NAME SPOT_ID  \n",
       "0             NM_017957           EPN3     NaN  \n",
       "1             NM_021235        EPS15L1     NaN  \n",
       "2  NM_005345, NM_005346  HSPA1A-HSPA1B     NaN  \n",
       "3             NM_016494       LOC51255     NaN  \n",
       "4             NM_178014     TUBB_HUMAN     NaN  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse78884 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse78884/GSE78884.top.table.tsv\", sep=\"\\t\")\n",
    "gse78884 = gse78884[gse78884[\"adj.P.Val\"] <= 0.05]\n",
    "gse78884_genes = sorted(list(set(gse78884[\"GENE_NAME\"].values.tolist())))\n",
    "with open(\"gse78884_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse78884_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "\n",
    "gse78884.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE89182"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>GI</th>\n",
       "      <th>Gene.symbol</th>\n",
       "      <th>Gene.title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ILMN_1860722</td>\n",
       "      <td>1.730000e-07</td>\n",
       "      <td>9.100000e-12</td>\n",
       "      <td>25.584537</td>\n",
       "      <td>16.441742</td>\n",
       "      <td>6.070218</td>\n",
       "      <td>23371365.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ILMN_1753342</td>\n",
       "      <td>1.740000e-07</td>\n",
       "      <td>1.830000e-11</td>\n",
       "      <td>24.099697</td>\n",
       "      <td>15.927773</td>\n",
       "      <td>4.138701</td>\n",
       "      <td>654823989.0</td>\n",
       "      <td>SAT1</td>\n",
       "      <td>spermidine/spermine N1-acetyltransferase 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ILMN_1793990</td>\n",
       "      <td>3.870000e-07</td>\n",
       "      <td>6.090000e-11</td>\n",
       "      <td>21.736955</td>\n",
       "      <td>15.000011</td>\n",
       "      <td>3.500998</td>\n",
       "      <td>33946335.0</td>\n",
       "      <td>ID2</td>\n",
       "      <td>inhibitor of DNA binding 2, HLH protein</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ILMN_2266948</td>\n",
       "      <td>1.540000e-06</td>\n",
       "      <td>3.240000e-10</td>\n",
       "      <td>18.815480</td>\n",
       "      <td>13.626231</td>\n",
       "      <td>2.717093</td>\n",
       "      <td>117168274.0</td>\n",
       "      <td>SLC38A1</td>\n",
       "      <td>solute carrier family 38 member 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ILMN_1660357</td>\n",
       "      <td>2.180000e-06</td>\n",
       "      <td>6.300000e-10</td>\n",
       "      <td>17.759279</td>\n",
       "      <td>13.055178</td>\n",
       "      <td>2.852408</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             ID     adj.P.Val       P.Value          t          B     logFC  \\\n",
       "0  ILMN_1860722  1.730000e-07  9.100000e-12  25.584537  16.441742  6.070218   \n",
       "1  ILMN_1753342  1.740000e-07  1.830000e-11  24.099697  15.927773  4.138701   \n",
       "2  ILMN_1793990  3.870000e-07  6.090000e-11  21.736955  15.000011  3.500998   \n",
       "3  ILMN_2266948  1.540000e-06  3.240000e-10  18.815480  13.626231  2.717093   \n",
       "4  ILMN_1660357  2.180000e-06  6.300000e-10  17.759279  13.055178  2.852408   \n",
       "\n",
       "            GI Gene.symbol                                  Gene.title  \n",
       "0   23371365.0         NaN                                         NaN  \n",
       "1  654823989.0        SAT1  spermidine/spermine N1-acetyltransferase 1  \n",
       "2   33946335.0         ID2     inhibitor of DNA binding 2, HLH protein  \n",
       "3  117168274.0     SLC38A1           solute carrier family 38 member 1  \n",
       "4          NaN         NaN                                         NaN  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse89182 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse89182/GSE89812.top.table.tsv\", sep=\"\\t\")\n",
    "gse89182 = gse89182[gse89182[\"adj.P.Val\"] <= 0.05]\n",
    "gse89182_genes = sorted(list(set([i for i in gse89182[\"Gene.symbol\"].values.tolist() if not \"nan\" in str(i).lower()])))\n",
    "with open(\"gse89182_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse89182_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "\n",
    "gse89182.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE125363"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>miRNA_ID</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Target</th>\n",
       "      <th>TargetID</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>55873</td>\n",
       "      <td>3.980000e-09</td>\n",
       "      <td>6.380000e-14</td>\n",
       "      <td>-10.325731</td>\n",
       "      <td>20.295798</td>\n",
       "      <td>-1.955485</td>\n",
       "      <td>hsa-mir-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0001541</td>\n",
       "      <td>ADAM10</td>\n",
       "      <td>102.0</td>\n",
       "      <td>Luciferase reporter assay//qRT-PCR//Western blot</td>\n",
       "      <td>27398144</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>55873</td>\n",
       "      <td>3.980000e-09</td>\n",
       "      <td>6.380000e-14</td>\n",
       "      <td>-10.325731</td>\n",
       "      <td>20.295798</td>\n",
       "      <td>-1.955485</td>\n",
       "      <td>hsa-mir-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0001541</td>\n",
       "      <td>ARHGAP1</td>\n",
       "      <td>392.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>23592263|21572407|20371350|27292025</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>55873</td>\n",
       "      <td>3.980000e-09</td>\n",
       "      <td>6.380000e-14</td>\n",
       "      <td>-10.325731</td>\n",
       "      <td>20.295798</td>\n",
       "      <td>-1.955485</td>\n",
       "      <td>hsa-mir-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0001541</td>\n",
       "      <td>ATP6V1E1</td>\n",
       "      <td>529.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>27292025</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>55873</td>\n",
       "      <td>3.980000e-09</td>\n",
       "      <td>6.380000e-14</td>\n",
       "      <td>-10.325731</td>\n",
       "      <td>20.295798</td>\n",
       "      <td>-1.955485</td>\n",
       "      <td>hsa-mir-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0001541</td>\n",
       "      <td>CCND1</td>\n",
       "      <td>595.0</td>\n",
       "      <td>Flow//Immunoblot//Luciferase reporter assay//q...</td>\n",
       "      <td>20948989|24248414</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>55873</td>\n",
       "      <td>3.980000e-09</td>\n",
       "      <td>6.380000e-14</td>\n",
       "      <td>-10.325731</td>\n",
       "      <td>20.295798</td>\n",
       "      <td>-1.955485</td>\n",
       "      <td>hsa-mir-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0001541</td>\n",
       "      <td>BCL2</td>\n",
       "      <td>596.0</td>\n",
       "      <td>ChIP-seq//Flow//Luciferase reporter assay//qRT...</td>\n",
       "      <td>21569010|24248414|24260067|24960059|27351886</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      ID     adj.P.Val       P.Value          t          B     logFC  \\\n",
       "0  55873  3.980000e-09  6.380000e-14 -10.325731  20.295798 -1.955485   \n",
       "1  55873  3.980000e-09  6.380000e-14 -10.325731  20.295798 -1.955485   \n",
       "2  55873  3.980000e-09  6.380000e-14 -10.325731  20.295798 -1.955485   \n",
       "3  55873  3.980000e-09  6.380000e-14 -10.325731  20.295798 -1.955485   \n",
       "4  55873  3.980000e-09  6.380000e-14 -10.325731  20.295798 -1.955485   \n",
       "\n",
       "    miRNA_ID SPOT_ID     Accession    Target  TargetID  \\\n",
       "0  hsa-mir-4     NaN  MIMAT0001541    ADAM10     102.0   \n",
       "1  hsa-mir-4     NaN  MIMAT0001541   ARHGAP1     392.0   \n",
       "2  hsa-mir-4     NaN  MIMAT0001541  ATP6V1E1     529.0   \n",
       "3  hsa-mir-4     NaN  MIMAT0001541     CCND1     595.0   \n",
       "4  hsa-mir-4     NaN  MIMAT0001541      BCL2     596.0   \n",
       "\n",
       "                                          Experiment  \\\n",
       "0   Luciferase reporter assay//qRT-PCR//Western blot   \n",
       "1                                           PAR-CLIP   \n",
       "2                                           PAR-CLIP   \n",
       "3  Flow//Immunoblot//Luciferase reporter assay//q...   \n",
       "4  ChIP-seq//Flow//Luciferase reporter assay//qRT...   \n",
       "\n",
       "                                     Literature       Tissue  \n",
       "0                                      27398144  Unspecified  \n",
       "1           23592263|21572407|20371350|27292025  Unspecified  \n",
       "2                                      27292025  Unspecified  \n",
       "3                             20948989|24248414  Unspecified  \n",
       "4  21569010|24248414|24260067|24960059|27351886  Unspecified  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse125363 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse125363/GSE125363.top.table.tsv\", sep=\"\\t\")\n",
    "gse125363 = gse125363[gse125363[\"adj.P.Val\"] <= 0.05]\n",
    "gene_mirmapping = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse125363/mir2gene.csv\")\n",
    "gene_mirmapping.columns = [\"miRNA_ID\"] + gene_mirmapping.columns.tolist()[1:]\n",
    "gene_mirmapping[\"miRNA_ID\"] = [i.lower()[:-3] for i in gene_mirmapping[\"miRNA_ID\"].values.tolist()]\n",
    "gse125363[\"miRNA_ID\"] = [str(i).lower()[:-3] for i in gse125363[\"miRNA_ID\"].values.tolist()]\n",
    "gse125363 = gse125363.merge(gene_mirmapping, on=[\"miRNA_ID\"])\n",
    "gse125363_genes = sorted(list(set(gse125363[\"Target\"].values.tolist())))\n",
    "with open(\"gse125363_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse125363_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "\n",
    "gse125363.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE125364 [Removed]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gse125364 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse125364/GSE125364.top.table.tsv\", sep=\"\\t\")\n",
    "# gse125364 = gse125364[gse125364[\"adj.P.Val\"] <= 0.05]\n",
    "# gene_probemapping = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse125364/gProfiler_hsapiens_28-04-2023_17-30-55.csv\")\n",
    "# gene_probemapping.columns = [\"SPOT_ID\"] + gene_probemapping.columns.tolist()[1:]\n",
    "# gse125364 = gse125364.merge(gene_probemapping, on=[\"SPOT_ID\"])\n",
    "# gse125364_genes = sorted(list(set(gse125364[\"name\"].values.tolist())))\n",
    "# with open(\"gse125364_genes.txt\", \"w\") as handle:\n",
    "#     for gen in gse125364_genes:\n",
    "#         handle.write(gen + \"\\n\")\n",
    "\n",
    "# gse125364.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE137369"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse137369_genes = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/gse137369_genes.txt\").read().split(\"\\n\")[:-1]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE 17306"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>ORGANISM</th>\n",
       "      <th>miRNA_ID</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Target</th>\n",
       "      <th>TargetID</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa-miR-663</td>\n",
       "      <td>0.000307</td>\n",
       "      <td>8.300000e-07</td>\n",
       "      <td>5.58432</td>\n",
       "      <td>5.6561</td>\n",
       "      <td>4.576052</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>27.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>24398324|23446348|21572407|20371350</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa-miR-663</td>\n",
       "      <td>0.000307</td>\n",
       "      <td>8.300000e-07</td>\n",
       "      <td>5.58432</td>\n",
       "      <td>5.6561</td>\n",
       "      <td>4.576052</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ACVR2B</td>\n",
       "      <td>93.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>21572407</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa-miR-663</td>\n",
       "      <td>0.000307</td>\n",
       "      <td>8.300000e-07</td>\n",
       "      <td>5.58432</td>\n",
       "      <td>5.6561</td>\n",
       "      <td>4.576052</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ADD2</td>\n",
       "      <td>119.0</td>\n",
       "      <td>CLASH</td>\n",
       "      <td>23622248</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa-miR-663</td>\n",
       "      <td>0.000307</td>\n",
       "      <td>8.300000e-07</td>\n",
       "      <td>5.58432</td>\n",
       "      <td>5.6561</td>\n",
       "      <td>4.576052</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ADORA3</td>\n",
       "      <td>140.0</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>19536157</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa-miR-663</td>\n",
       "      <td>0.000307</td>\n",
       "      <td>8.300000e-07</td>\n",
       "      <td>5.58432</td>\n",
       "      <td>5.6561</td>\n",
       "      <td>4.576052</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ALDH3B1</td>\n",
       "      <td>221.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>23592263</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID  adj.P.Val       P.Value        t       B     logFC  \\\n",
       "0  hsa-miR-663   0.000307  8.300000e-07  5.58432  5.6561  4.576052   \n",
       "1  hsa-miR-663   0.000307  8.300000e-07  5.58432  5.6561  4.576052   \n",
       "2  hsa-miR-663   0.000307  8.300000e-07  5.58432  5.6561  4.576052   \n",
       "3  hsa-miR-663   0.000307  8.300000e-07  5.58432  5.6561  4.576052   \n",
       "4  hsa-miR-663   0.000307  8.300000e-07  5.58432  5.6561  4.576052   \n",
       "\n",
       "       ORGANISM  miRNA_ID SPOT_ID     Accession   Target  TargetID Experiment  \\\n",
       "0  Homo sapiens  hsa-mir-     NaN  MIMAT0000104     ABL2      27.0   PAR-CLIP   \n",
       "1  Homo sapiens  hsa-mir-     NaN  MIMAT0000104   ACVR2B      93.0   PAR-CLIP   \n",
       "2  Homo sapiens  hsa-mir-     NaN  MIMAT0000104     ADD2     119.0      CLASH   \n",
       "3  Homo sapiens  hsa-mir-     NaN  MIMAT0000104   ADORA3     140.0  HITS-CLIP   \n",
       "4  Homo sapiens  hsa-mir-     NaN  MIMAT0000104  ALDH3B1     221.0   PAR-CLIP   \n",
       "\n",
       "                            Literature       Tissue  \n",
       "0  24398324|23446348|21572407|20371350  Unspecified  \n",
       "1                             21572407  Unspecified  \n",
       "2                             23622248  Unspecified  \n",
       "3                             19536157  Unspecified  \n",
       "4                             23592263  Unspecified  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse17306 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse17306/GSE17306.top.table.tsv\", sep=\"\\t\")\n",
    "gse17306 = gse17306[gse17306[\"adj.P.Val\"] <= 0.05]\n",
    "gene_mirmapping = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse17306/mir2gene.csv\")\n",
    "gene_mirmapping.columns = [\"miRNA_ID\"] + gene_mirmapping.columns.tolist()[1:]\n",
    "gene_mirmapping[\"miRNA_ID\"] = [i.lower()[:-3] for i in gene_mirmapping[\"miRNA_ID\"].values.tolist()]\n",
    "gse17306[\"miRNA_ID\"] = [str(i).lower()[:-3] for i in gse17306[\"miRNA_ID\"].values.tolist()]\n",
    "gse17306 = gse17306.merge(gene_mirmapping, on=[\"miRNA_ID\"])\n",
    "gse17306_genes = sorted(list(set(gse17306[\"Target\"].values.tolist())))\n",
    "with open(\"gse17306_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse17306_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "\n",
    "gse17306.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE16558"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse16558 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse16558/GSE16558.top.table.tsv\", sep=\"\\t\")\n",
    "gse16558 = gse16558[gse16558[\"adj.P.Val\"] <= 0.05]\n",
    "gse16558_gene = sorted(list(set([i for i in gse16558[\"Gene.symbol\"].values.tolist() if not \"nan\" in str(i).lower()])))\n",
    "with open(\"gse16558_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse16558_gene:\n",
    "        handle.write(gen + \"\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE17498"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>ORGANISM</th>\n",
       "      <th>miRNA_ID</th>\n",
       "      <th>SPOT_ID</th>\n",
       "      <th>SPOT_ID.1</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Target</th>\n",
       "      <th>TargetID</th>\n",
       "      <th>Experiment</th>\n",
       "      <th>Literature</th>\n",
       "      <th>Tissue</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsa-miR-363</td>\n",
       "      <td>2.360000e-13</td>\n",
       "      <td>3.260000e-16</td>\n",
       "      <td>-12.670357</td>\n",
       "      <td>26.726</td>\n",
       "      <td>-3.319857</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ABL2</td>\n",
       "      <td>27.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>24398324|23446348|21572407|20371350</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsa-miR-363</td>\n",
       "      <td>2.360000e-13</td>\n",
       "      <td>3.260000e-16</td>\n",
       "      <td>-12.670357</td>\n",
       "      <td>26.726</td>\n",
       "      <td>-3.319857</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ACVR2B</td>\n",
       "      <td>93.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>21572407</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsa-miR-363</td>\n",
       "      <td>2.360000e-13</td>\n",
       "      <td>3.260000e-16</td>\n",
       "      <td>-12.670357</td>\n",
       "      <td>26.726</td>\n",
       "      <td>-3.319857</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ADD2</td>\n",
       "      <td>119.0</td>\n",
       "      <td>CLASH</td>\n",
       "      <td>23622248</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsa-miR-363</td>\n",
       "      <td>2.360000e-13</td>\n",
       "      <td>3.260000e-16</td>\n",
       "      <td>-12.670357</td>\n",
       "      <td>26.726</td>\n",
       "      <td>-3.319857</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ADORA3</td>\n",
       "      <td>140.0</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>19536157</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsa-miR-363</td>\n",
       "      <td>2.360000e-13</td>\n",
       "      <td>3.260000e-16</td>\n",
       "      <td>-12.670357</td>\n",
       "      <td>26.726</td>\n",
       "      <td>-3.319857</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>hsa-mir-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIMAT0000104</td>\n",
       "      <td>ALDH3B1</td>\n",
       "      <td>221.0</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>23592263</td>\n",
       "      <td>Unspecified</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID     adj.P.Val       P.Value          t       B     logFC  \\\n",
       "0  hsa-miR-363  2.360000e-13  3.260000e-16 -12.670357  26.726 -3.319857   \n",
       "1  hsa-miR-363  2.360000e-13  3.260000e-16 -12.670357  26.726 -3.319857   \n",
       "2  hsa-miR-363  2.360000e-13  3.260000e-16 -12.670357  26.726 -3.319857   \n",
       "3  hsa-miR-363  2.360000e-13  3.260000e-16 -12.670357  26.726 -3.319857   \n",
       "4  hsa-miR-363  2.360000e-13  3.260000e-16 -12.670357  26.726 -3.319857   \n",
       "\n",
       "       ORGANISM  miRNA_ID  SPOT_ID  SPOT_ID.1     Accession   Target  \\\n",
       "0  Homo sapiens  hsa-mir-      NaN        NaN  MIMAT0000104     ABL2   \n",
       "1  Homo sapiens  hsa-mir-      NaN        NaN  MIMAT0000104   ACVR2B   \n",
       "2  Homo sapiens  hsa-mir-      NaN        NaN  MIMAT0000104     ADD2   \n",
       "3  Homo sapiens  hsa-mir-      NaN        NaN  MIMAT0000104   ADORA3   \n",
       "4  Homo sapiens  hsa-mir-      NaN        NaN  MIMAT0000104  ALDH3B1   \n",
       "\n",
       "   TargetID Experiment                           Literature       Tissue  \n",
       "0      27.0   PAR-CLIP  24398324|23446348|21572407|20371350  Unspecified  \n",
       "1      93.0   PAR-CLIP                             21572407  Unspecified  \n",
       "2     119.0      CLASH                             23622248  Unspecified  \n",
       "3     140.0  HITS-CLIP                             19536157  Unspecified  \n",
       "4     221.0   PAR-CLIP                             23592263  Unspecified  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse17498 = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse17498/GSE17498.top.table.tsv\", sep=\"\\t\")\n",
    "gse17498 = gse17498[gse17498[\"adj.P.Val\"] <= 0.05]\n",
    "gene_mirmapping = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse17498/mir2gene.csv\")\n",
    "gene_mirmapping.columns = [\"miRNA_ID\"] + gene_mirmapping.columns.tolist()[1:]\n",
    "gene_mirmapping[\"miRNA_ID\"] = [i.lower()[:-3] for i in gene_mirmapping[\"miRNA_ID\"].values.tolist()]\n",
    "gse17498[\"miRNA_ID\"] = [i.lower()[:-3] for i in gse17498[\"miRNA_ID\"].values.tolist()]\n",
    "gse17498 = gse17498.merge(gene_mirmapping, on=[\"miRNA_ID\"])\n",
    "gse17498_genes = sorted(list(set(gse17498[\"Target\"].values.tolist())))\n",
    "with open(\"gse17498_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse17498_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "\n",
    "gse17498.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE146649 [NOT PUBLISHED]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = pd.read_csv(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse146649/GSE146649.top.table.tsv\", sep=\"\\t\")\n",
    "# genelist = []\n",
    "# for idx in tqdm(range(df.shape[0])):\n",
    "#     if not '//' in str(df.iloc[idx, -2]):\n",
    "#         genelist.append(df.iloc[idx, -2])    \n",
    "#     else:\n",
    "#         glist = df.iloc[idx, -2].split(\"//\")\n",
    "#         for i in glist:\n",
    "#             genelist.append(i)\n",
    "# genelist = sorted([i[1:] for i in genelist if not str(i)[0] == \"//\" and not \"nan\" in str(i).lower()])\n",
    "# with open(\"gse146649_genes.txt\", \"w\") as handle:\n",
    "#     for gen in genelist:\n",
    "#         handle.write(gen + \"\\n\")"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE149833"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>logFC</th>\n",
       "      <th>gene_assignment</th>\n",
       "      <th>SPOT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TC1200010158.hg.1</td>\n",
       "      <td>0.00274</td>\n",
       "      <td>1.970000e-08</td>\n",
       "      <td>-8.031878</td>\n",
       "      <td>-2.900</td>\n",
       "      <td>NM_004985 // KRAS // Kirsten rat sarcoma viral...</td>\n",
       "      <td>NM_004985 // RefSeq</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TC0500010576.hg.1</td>\n",
       "      <td>0.01032</td>\n",
       "      <td>1.490000e-07</td>\n",
       "      <td>7.173916</td>\n",
       "      <td>0.778</td>\n",
       "      <td>---</td>\n",
       "      <td>T281415 // miTranscriptome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TC1700008969.hg.1</td>\n",
       "      <td>0.02887</td>\n",
       "      <td>6.240000e-07</td>\n",
       "      <td>6.587241</td>\n",
       "      <td>0.695</td>\n",
       "      <td>uc060knl.1 // RNU6-97P // RNA, U6 small nuclea...</td>\n",
       "      <td>uc060knl.1 // UCSC Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TC0600009004.hg.1</td>\n",
       "      <td>0.05402</td>\n",
       "      <td>1.570000e-06</td>\n",
       "      <td>6.218865</td>\n",
       "      <td>0.501</td>\n",
       "      <td>uc063qka.1 // RNU6-1299P // RNA, U6 small nucl...</td>\n",
       "      <td>uc063qka.1 // UCSC Genes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TC0600012849.hg.1</td>\n",
       "      <td>0.05402</td>\n",
       "      <td>2.400000e-06</td>\n",
       "      <td>6.049707</td>\n",
       "      <td>0.640</td>\n",
       "      <td>uc063qqg.1 // RNU6-906P // RNA, U6 small nucle...</td>\n",
       "      <td>uc063qqg.1 // UCSC Genes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID  adj.P.Val       P.Value         t  logFC  \\\n",
       "0  TC1200010158.hg.1    0.00274  1.970000e-08 -8.031878 -2.900   \n",
       "1  TC0500010576.hg.1    0.01032  1.490000e-07  7.173916  0.778   \n",
       "2  TC1700008969.hg.1    0.02887  6.240000e-07  6.587241  0.695   \n",
       "3  TC0600009004.hg.1    0.05402  1.570000e-06  6.218865  0.501   \n",
       "4  TC0600012849.hg.1    0.05402  2.400000e-06  6.049707  0.640   \n",
       "\n",
       "                                     gene_assignment  \\\n",
       "0  NM_004985 // KRAS // Kirsten rat sarcoma viral...   \n",
       "1                                                ---   \n",
       "2  uc060knl.1 // RNU6-97P // RNA, U6 small nuclea...   \n",
       "3  uc063qka.1 // RNU6-1299P // RNA, U6 small nucl...   \n",
       "4  uc063qqg.1 // RNU6-906P // RNA, U6 small nucle...   \n",
       "\n",
       "                      SPOT_ID  \n",
       "0         NM_004985 // RefSeq  \n",
       "1  T281415 // miTranscriptome  \n",
       "2    uc060knl.1 // UCSC Genes  \n",
       "3    uc063qka.1 // UCSC Genes  \n",
       "4    uc063qqg.1 // UCSC Genes  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel('/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse149833/GSE149833.top.table.xlsx')\n",
    "gse149833_genes = sorted([i.split(\"//\")[1].replace(' ','') for i in df[\"gene_assignment\"].values.tolist() if not \"nan\" in str(i).lower() and not \"--\" in i])\n",
    "with open(\"gse149833_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse149833_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "df.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GSE167042"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>t</th>\n",
       "      <th>B</th>\n",
       "      <th>logFC</th>\n",
       "      <th>probeset_id</th>\n",
       "      <th>seqname</th>\n",
       "      <th>strand</th>\n",
       "      <th>start</th>\n",
       "      <th>total_probes</th>\n",
       "      <th>gene_assignment</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TC0500007076.hg.1</td>\n",
       "      <td>0.0323</td>\n",
       "      <td>3.810000e-07</td>\n",
       "      <td>18.7</td>\n",
       "      <td>5.169442</td>\n",
       "      <td>3.56</td>\n",
       "      <td>TC0500007076.hg.1</td>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>32653048</td>\n",
       "      <td>10.0</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TC1700007186.hg.1</td>\n",
       "      <td>0.0323</td>\n",
       "      <td>4.650000e-07</td>\n",
       "      <td>18.2</td>\n",
       "      <td>5.090728</td>\n",
       "      <td>3.60</td>\n",
       "      <td>TC1700007186.hg.1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>+</td>\n",
       "      <td>19603797</td>\n",
       "      <td>10.0</td>\n",
       "      <td>ENST00000417560 // MTND2P12 // mitochondrially...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TC0500007077.hg.1</td>\n",
       "      <td>0.0346</td>\n",
       "      <td>7.480000e-07</td>\n",
       "      <td>17.0</td>\n",
       "      <td>4.891762</td>\n",
       "      <td>3.18</td>\n",
       "      <td>TC0500007077.hg.1</td>\n",
       "      <td>chr5</td>\n",
       "      <td>+</td>\n",
       "      <td>32689070</td>\n",
       "      <td>10.0</td>\n",
       "      <td>NM_000908 // NPR3 // natriuretic peptide recep...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TC1700007184.hg.1</td>\n",
       "      <td>0.0489</td>\n",
       "      <td>1.410000e-06</td>\n",
       "      <td>15.4</td>\n",
       "      <td>4.597955</td>\n",
       "      <td>3.38</td>\n",
       "      <td>TC1700007184.hg.1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>+</td>\n",
       "      <td>19602259</td>\n",
       "      <td>10.0</td>\n",
       "      <td>ENST00000427400 // MTND1P14 // mitochondrially...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TC1000008894.hg.1</td>\n",
       "      <td>0.0530</td>\n",
       "      <td>2.040000e-06</td>\n",
       "      <td>14.6</td>\n",
       "      <td>4.411300</td>\n",
       "      <td>3.45</td>\n",
       "      <td>TC1000008894.hg.1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>+</td>\n",
       "      <td>110545655</td>\n",
       "      <td>10.0</td>\n",
       "      <td>---</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID  adj.P.Val       P.Value     t         B  logFC  \\\n",
       "0  TC0500007076.hg.1     0.0323  3.810000e-07  18.7  5.169442   3.56   \n",
       "1  TC1700007186.hg.1     0.0323  4.650000e-07  18.2  5.090728   3.60   \n",
       "2  TC0500007077.hg.1     0.0346  7.480000e-07  17.0  4.891762   3.18   \n",
       "3  TC1700007184.hg.1     0.0489  1.410000e-06  15.4  4.597955   3.38   \n",
       "4  TC1000008894.hg.1     0.0530  2.040000e-06  14.6  4.411300   3.45   \n",
       "\n",
       "         probeset_id seqname strand      start  total_probes  \\\n",
       "0  TC0500007076.hg.1    chr5      +   32653048          10.0   \n",
       "1  TC1700007186.hg.1   chr17      +   19603797          10.0   \n",
       "2  TC0500007077.hg.1    chr5      +   32689070          10.0   \n",
       "3  TC1700007184.hg.1   chr17      +   19602259          10.0   \n",
       "4  TC1000008894.hg.1   chr10      +  110545655          10.0   \n",
       "\n",
       "                                     gene_assignment  \n",
       "0                                                ---  \n",
       "1  ENST00000417560 // MTND2P12 // mitochondrially...  \n",
       "2  NM_000908 // NPR3 // natriuretic peptide recep...  \n",
       "3  ENST00000427400 // MTND1P14 // mitochondrially...  \n",
       "4                                                ---  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_excel(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/additional_database_validation/additional_database_validation/gse167042/GSE167042.top.table.xlsx\")\n",
    "gse167042_genes = sorted([i.split(\"//\")[1].replace(' ','') for i in df[\"gene_assignment\"].values.tolist() if not \"nan\" in str(i).lower() and not \"--\" in i])\n",
    "with open(\"gse167042_genes.txt\", \"w\") as handle:\n",
    "    for gen in gse167042_genes:\n",
    "        handle.write(gen + \"\\n\")\n",
    "df.head()"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Consensus or Union of all dstasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "common_genes = list(set(gse37302_genes) & set(gse39683_genecounts1_genes) & set(gse17306_genes) & set(gse16558_gene) & set(gse72213_genes) & set(gse17498_genes) & set(gse89182_genes) & set(gse125363_genes) & set(gse137369_genes) & set(gse167042_genes) & set(gse149833_genes))\n",
    "common_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_genes = list(set(gse37302_genes + gse39683_genecounts1_genes + gse17306_genes + gse16558_gene + gse72213_genes + gse17498_genes + gse89182_genes + gse125363_genes + gse137369_genes + gse167042_genes + gse149833_genes))\n",
    "top500_genes = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/ppi_string_merged/new_model4/best_performing_model/top_500_genes.txt\").read().split(\"\\n\")\n",
    "top100_genes = top500_genes[:100]\n",
    "top250_genes = top500_genes[:250]\n",
    "genepanel77 = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/farcastbio/77genelist.txt\").read().split(\"\\n\")\n",
    "genepanel34 = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/farcastbio/genelist77_mm_mgus.txt\").read().split(\"\\n\")[:-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 13%|        | 107/798 [00:00<00:00, 1062.58it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|| 798/798 [00:00<00:00, 925.29it/s] \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>Rank</th>\n",
       "      <th>GSE37302</th>\n",
       "      <th>GSE39683</th>\n",
       "      <th>GSE17306</th>\n",
       "      <th>GSE16558</th>\n",
       "      <th>GSE72213</th>\n",
       "      <th>GSE17498</th>\n",
       "      <th>GSE89182</th>\n",
       "      <th>GSE125363</th>\n",
       "      <th>GSE149833</th>\n",
       "      <th>GSE167042</th>\n",
       "      <th>GSE137369</th>\n",
       "      <th>Total_validated_datasets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MUC6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LILRA1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LILRB1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Genes  Rank  GSE37302  GSE39683  GSE17306  GSE16558  GSE72213  GSE17498  \\\n",
       "0      MUC6     1         0         0         1         0         0         1   \n",
       "1    LILRA1     2         0         0         0         0         0         0   \n",
       "2    LILRB1     3         0         0         1         0         0         1   \n",
       "3   PDE4DIP     4         1         0         1         1         0         1   \n",
       "4  EIF4EBP1     5         1         0         1         0         0         1   \n",
       "\n",
       "   GSE89182  GSE125363  GSE149833  GSE167042  GSE137369  \\\n",
       "0         0          1          1          0          0   \n",
       "1         0          1          1          0          0   \n",
       "2         1          1          0          1          0   \n",
       "3         0          1          0          0          0   \n",
       "4         1          1          0          1          0   \n",
       "\n",
       "   Total_validated_datasets  \n",
       "0                         4  \n",
       "1                         2  \n",
       "2                         5  \n",
       "3                         5  \n",
       "4                         6  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene798_rank =  pd.read_excel(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/ppi_string_merged/new_model4/best_performing_model/gene798_ranking.xlsx\", index_col=0)\n",
    "df_dataset_counts = pd.DataFrame()\n",
    "gse37302_col, gse39683_genecounts1_col, gse17306_col, gse16558_col, gse72213_col, gse17498_col, gse89182_col, gse125363_col, gse167042_col, gse149833_col, gse137369_col = [], [],[],[],[],[],[],[],[],[], []\n",
    "generank = []\n",
    "genelist = gene798_rank.index.tolist()\n",
    "for gen in tqdm(genelist):\n",
    "    generank.append(genelist.index(gen)+1)\n",
    "    if gen in gse37302_genes:\n",
    "        gse37302_col.append(1)\n",
    "    else:\n",
    "        gse37302_col.append(0)\n",
    "\n",
    "    if gen in gse39683_genecounts1_genes:\n",
    "        gse39683_genecounts1_col.append(1)\n",
    "    else:\n",
    "        gse39683_genecounts1_col.append(0)\n",
    "\n",
    "    if gen in gse17306_genes:\n",
    "        gse17306_col.append(1)\n",
    "    else:\n",
    "        gse17306_col.append(0)\n",
    "\n",
    "    if gen in gse16558_gene:\n",
    "        gse16558_col.append(1)\n",
    "    else:\n",
    "        gse16558_col.append(0)\n",
    "\n",
    "    if gen in gse72213_genes:\n",
    "        gse72213_col.append(1)\n",
    "    else:\n",
    "        gse72213_col.append(0)\n",
    "\n",
    "    if gen in gse17498_genes:\n",
    "        gse17498_col.append(1)\n",
    "    else:\n",
    "        gse17498_col.append(0)\n",
    "\n",
    "    if gen in gse89182_genes:\n",
    "        gse89182_col.append(1)\n",
    "    else:\n",
    "        gse89182_col.append(0)\n",
    "\n",
    "    if gen in gse125363_genes:\n",
    "        gse125363_col.append(1)\n",
    "    else:\n",
    "        gse125363_col.append(0)\n",
    "\n",
    "    if gen in gse167042_genes:\n",
    "        gse167042_col.append(1)\n",
    "    else:\n",
    "        gse167042_col.append(0)\n",
    "        \n",
    "    if gen in gse149833_genes:\n",
    "        gse149833_col.append(1)\n",
    "    else:\n",
    "        gse149833_col.append(0)\n",
    "\n",
    "    if gen in gse137369_genes:\n",
    "        gse137369_col.append(1)\n",
    "    else:\n",
    "        gse137369_col.append(0)\n",
    "\n",
    "df_dataset_counts['Genes'] = genelist\n",
    "df_dataset_counts['Rank'] = generank\n",
    "df_dataset_counts['GSE37302'] = gse37302_col\n",
    "df_dataset_counts['GSE39683'] = gse39683_genecounts1_col\n",
    "df_dataset_counts['GSE17306'] = gse17306_col\n",
    "df_dataset_counts['GSE16558'] = gse16558_col\n",
    "df_dataset_counts['GSE72213'] = gse72213_col\n",
    "df_dataset_counts['GSE17498'] = gse17498_col\n",
    "df_dataset_counts['GSE89182'] = gse89182_col\n",
    "df_dataset_counts['GSE125363'] = gse125363_col\n",
    "df_dataset_counts['GSE149833'] = gse149833_col\n",
    "df_dataset_counts['GSE167042'] = gse167042_col\n",
    "df_dataset_counts['GSE137369'] = gse137369_col\n",
    "df_dataset_counts[\"Total_validated_datasets\"] = [sum(df_dataset_counts.loc[idx,:].values.tolist()[2:]) for idx in range(df_dataset_counts.shape[0])]\n",
    "\n",
    "df_dataset_counts.to_excel('gene798_dataset_counts.xlsx', index=False, encoding=\"utf=8\")\n",
    "df_dataset_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/500 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|| 500/500 [00:00<00:00, 918.57it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>Rank</th>\n",
       "      <th>GSE37302</th>\n",
       "      <th>GSE39683</th>\n",
       "      <th>GSE17306</th>\n",
       "      <th>GSE16558</th>\n",
       "      <th>GSE72213</th>\n",
       "      <th>GSE17498</th>\n",
       "      <th>GSE89182</th>\n",
       "      <th>GSE125363</th>\n",
       "      <th>GSE149833</th>\n",
       "      <th>GSE167042</th>\n",
       "      <th>GSE137369</th>\n",
       "      <th>Total_validated_datasets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MUC6</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>LILRA1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>LILRB1</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PDE4DIP</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>EIF4EBP1</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Genes  Rank  GSE37302  GSE39683  GSE17306  GSE16558  GSE72213  GSE17498  \\\n",
       "0      MUC6     1         0         0         1         0         0         1   \n",
       "1    LILRA1     2         0         0         0         0         0         0   \n",
       "2    LILRB1     3         0         0         1         0         0         1   \n",
       "3   PDE4DIP     4         1         0         1         1         0         1   \n",
       "4  EIF4EBP1     5         1         0         1         0         0         1   \n",
       "\n",
       "   GSE89182  GSE125363  GSE149833  GSE167042  GSE137369  \\\n",
       "0         0          1          1          0          0   \n",
       "1         0          1          1          0          0   \n",
       "2         1          1          0          1          0   \n",
       "3         0          1          0          0          0   \n",
       "4         1          1          0          1          0   \n",
       "\n",
       "   Total_validated_datasets  \n",
       "0                         4  \n",
       "1                         2  \n",
       "2                         5  \n",
       "3                         5  \n",
       "4                         6  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dataset_counts = pd.DataFrame()\n",
    "gse37302_col, gse39683_genecounts1_col, gse17306_col, gse16558_col, gse72213_col, gse17498_col, gse89182_col, gse125363_col, gse167042_col, gse149833_col, gse137369_col = [], [],[],[],[],[],[],[],[],[], []\n",
    "generank = []\n",
    "genelist = top500_genes\n",
    "for gen in tqdm(genelist):\n",
    "    generank.append(genelist.index(gen)+1)\n",
    "    if gen in gse37302_genes:\n",
    "        gse37302_col.append(1)\n",
    "    else:\n",
    "        gse37302_col.append(0)\n",
    "\n",
    "    if gen in gse39683_genecounts1_genes:\n",
    "        gse39683_genecounts1_col.append(1)\n",
    "    else:\n",
    "        gse39683_genecounts1_col.append(0)\n",
    "\n",
    "    if gen in gse17306_genes:\n",
    "        gse17306_col.append(1)\n",
    "    else:\n",
    "        gse17306_col.append(0)\n",
    "\n",
    "    if gen in gse16558_gene:\n",
    "        gse16558_col.append(1)\n",
    "    else:\n",
    "        gse16558_col.append(0)\n",
    "\n",
    "    if gen in gse72213_genes:\n",
    "        gse72213_col.append(1)\n",
    "    else:\n",
    "        gse72213_col.append(0)\n",
    "\n",
    "    if gen in gse17498_genes:\n",
    "        gse17498_col.append(1)\n",
    "    else:\n",
    "        gse17498_col.append(0)\n",
    "\n",
    "    if gen in gse89182_genes:\n",
    "        gse89182_col.append(1)\n",
    "    else:\n",
    "        gse89182_col.append(0)\n",
    "\n",
    "    if gen in gse125363_genes:\n",
    "        gse125363_col.append(1)\n",
    "    else:\n",
    "        gse125363_col.append(0)\n",
    "\n",
    "    if gen in gse167042_genes:\n",
    "        gse167042_col.append(1)\n",
    "    else:\n",
    "        gse167042_col.append(0)\n",
    "        \n",
    "    if gen in gse149833_genes:\n",
    "        gse149833_col.append(1)\n",
    "    else:\n",
    "        gse149833_col.append(0)\n",
    "\n",
    "    if gen in gse137369_genes:\n",
    "        gse137369_col.append(1)\n",
    "    else:\n",
    "        gse137369_col.append(0)\n",
    "\n",
    "df_dataset_counts['Genes'] = genelist\n",
    "df_dataset_counts['Rank'] = generank\n",
    "df_dataset_counts['GSE37302'] = gse37302_col\n",
    "df_dataset_counts['GSE39683'] = gse39683_genecounts1_col\n",
    "df_dataset_counts['GSE17306'] = gse17306_col\n",
    "df_dataset_counts['GSE16558'] = gse16558_col\n",
    "df_dataset_counts['GSE72213'] = gse72213_col\n",
    "df_dataset_counts['GSE17498'] = gse17498_col\n",
    "df_dataset_counts['GSE89182'] = gse89182_col\n",
    "df_dataset_counts['GSE125363'] = gse125363_col\n",
    "df_dataset_counts['GSE149833'] = gse149833_col\n",
    "df_dataset_counts['GSE167042'] = gse167042_col\n",
    "df_dataset_counts['GSE137369'] = gse137369_col\n",
    "df_dataset_counts[\"Total_validated_datasets\"] = [sum(df_dataset_counts.loc[idx,:].values.tolist()[2:]) for idx in range(df_dataset_counts.shape[0])]\n",
    "\n",
    "df_dataset_counts.to_excel('top500_dataset_counts.xlsx', index=False, encoding=\"utf=8\")\n",
    "df_dataset_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/701 [00:00<?, ?it/s]"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|| 701/701 [00:00<00:00, 1009.12it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genes</th>\n",
       "      <th>Rank</th>\n",
       "      <th>GSE37302</th>\n",
       "      <th>GSE39683</th>\n",
       "      <th>GSE17306</th>\n",
       "      <th>GSE16558</th>\n",
       "      <th>GSE72213</th>\n",
       "      <th>GSE17498</th>\n",
       "      <th>GSE89182</th>\n",
       "      <th>GSE125363</th>\n",
       "      <th>GSE149833</th>\n",
       "      <th>GSE167042</th>\n",
       "      <th>GSE137369</th>\n",
       "      <th>Total_validated_datasets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCA3</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Genes  Rank  GSE37302  GSE39683  GSE17306  GSE16558  GSE72213  GSE17498  \\\n",
       "0  ABCA1     1         1         0         1         0         0         1   \n",
       "1  ABCA3     2         0         0         1         0         0         1   \n",
       "2  ABCA7     3         0         0         1         0         0         1   \n",
       "3   ABL2     4         0         0         1         0         0         1   \n",
       "4  ACACB     5         0         0         1         0         0         1   \n",
       "\n",
       "   GSE89182  GSE125363  GSE149833  GSE167042  GSE137369  \\\n",
       "0         0          1          1          0          0   \n",
       "1         0          1          1          0          0   \n",
       "2         0          1          1          1          0   \n",
       "3         0          1          0          0          0   \n",
       "4         1          1          0          1          0   \n",
       "\n",
       "   Total_validated_datasets  \n",
       "0                         5  \n",
       "1                         4  \n",
       "2                         5  \n",
       "3                         3  \n",
       "4                         5  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_dataset_counts = pd.DataFrame()\n",
    "gene701 = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene701/gene701.txt\").read().split(\"\\n\")\n",
    "gene701 = [i for i in gene701 if i]\n",
    "gse37302_col, gse39683_genecounts1_col, gse17306_col, gse16558_col, gse72213_col, gse17498_col, gse89182_col, gse125363_col, gse167042_col, gse149833_col, gse137369_col = [], [],[],[],[],[],[],[],[],[], []\n",
    "generank = []\n",
    "genelist = gene701\n",
    "for gen in tqdm(genelist):\n",
    "    generank.append(genelist.index(gen)+1)\n",
    "    if gen in gse37302_genes:\n",
    "        gse37302_col.append(1)\n",
    "    else:\n",
    "        gse37302_col.append(0)\n",
    "\n",
    "    if gen in gse39683_genecounts1_genes:\n",
    "        gse39683_genecounts1_col.append(1)\n",
    "    else:\n",
    "        gse39683_genecounts1_col.append(0)\n",
    "\n",
    "    if gen in gse17306_genes:\n",
    "        gse17306_col.append(1)\n",
    "    else:\n",
    "        gse17306_col.append(0)\n",
    "\n",
    "    if gen in gse16558_gene:\n",
    "        gse16558_col.append(1)\n",
    "    else:\n",
    "        gse16558_col.append(0)\n",
    "\n",
    "    if gen in gse72213_genes:\n",
    "        gse72213_col.append(1)\n",
    "    else:\n",
    "        gse72213_col.append(0)\n",
    "\n",
    "    if gen in gse17498_genes:\n",
    "        gse17498_col.append(1)\n",
    "    else:\n",
    "        gse17498_col.append(0)\n",
    "\n",
    "    if gen in gse89182_genes:\n",
    "        gse89182_col.append(1)\n",
    "    else:\n",
    "        gse89182_col.append(0)\n",
    "\n",
    "    if gen in gse125363_genes:\n",
    "        gse125363_col.append(1)\n",
    "    else:\n",
    "        gse125363_col.append(0)\n",
    "\n",
    "    if gen in gse167042_genes:\n",
    "        gse167042_col.append(1)\n",
    "    else:\n",
    "        gse167042_col.append(0)\n",
    "        \n",
    "    if gen in gse149833_genes:\n",
    "        gse149833_col.append(1)\n",
    "    else:\n",
    "        gse149833_col.append(0)\n",
    "\n",
    "    if gen in gse137369_genes:\n",
    "        gse137369_col.append(1)\n",
    "    else:\n",
    "        gse137369_col.append(0)\n",
    "\n",
    "df_dataset_counts['Genes'] = genelist\n",
    "df_dataset_counts['Rank'] = generank\n",
    "df_dataset_counts['GSE37302'] = gse37302_col\n",
    "df_dataset_counts['GSE39683'] = gse39683_genecounts1_col\n",
    "df_dataset_counts['GSE17306'] = gse17306_col\n",
    "df_dataset_counts['GSE16558'] = gse16558_col\n",
    "df_dataset_counts['GSE72213'] = gse72213_col\n",
    "df_dataset_counts['GSE17498'] = gse17498_col\n",
    "df_dataset_counts['GSE89182'] = gse89182_col\n",
    "df_dataset_counts['GSE125363'] = gse125363_col\n",
    "df_dataset_counts['GSE149833'] = gse149833_col\n",
    "df_dataset_counts['GSE167042'] = gse167042_col\n",
    "df_dataset_counts['GSE137369'] = gse137369_col\n",
    "df_dataset_counts[\"Total_validated_datasets\"] = [sum(df_dataset_counts.loc[idx,:].values.tolist()[2:]) for idx in range(df_dataset_counts.shape[0])]\n",
    "\n",
    "df_dataset_counts.to_excel('/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene701/top701_dataset_counts.xlsx', index=False, encoding=\"utf=8\")\n",
    "df_dataset_counts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "742bfade6c7c4e9a7580832fc255028e6cb2f8f83ccad2220a0613772e136a55"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
