{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import pickle\n",
    "from tqdm import tqdm\n",
    "import ast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "mmrf_muse_path = \"/home/vivek/mmrf_data/mmrf_variant_callers/hg19/muse/annovar/vcf_with_AF/vcf_reheader/fathmm/filtered_vcf/\"\n",
    "mmrf_mutect_path = \"/home/vivek/mmrf_data/mmrf_variant_callers/hg19/mutect2/annovar/vcf_reheader/fathmm/filtered_vcf\"\n",
    "mmrf_sniper_path = \"/home/vivek/mmrf_data/mmrf_variant_callers/hg19/somatic_sniper/annovar/vcf_with_AF/vcf_reheader/fathmm/filtered_vcf\"\n",
    "mmrf_varscan_path = \"/home/vivek/mmrf_data/mmrf_variant_callers/hg19/varscan2/annovar/vcf_reheader/fathmm/filtered_vcf\"\n",
    "\n",
    "ega_muse_path = \"/home/vivek/ega_data_1901/muse/fathmm_filtered/filtered_vcf\"\n",
    "ega_mutect_path = \"/home/vivek/ega_data_1901/MUTECT2_analysis/fathmm_filtered/filtered_vcf\"\n",
    "ega_sniper_path = \"/home/vivek/ega_data_1901/SNIPER_analysis/fathmm_filtered/filtered_vcf\"\n",
    "ega_varscan_path = \"/home/vivek/ega_data_1901/varscan2/fathmm_filtered/filtered_vcf\"\n",
    "\n",
    "aiims_mm_muse_path = \"/home/vivek/aiims_data_processing/muse/mm_sample_analysis/fathmm_filtered/filtered_vcf\"\n",
    "aiims_mm_mutect_path = \"/home/vivek/aiims_data_processing/MUTECT2/mm_sample_analysis/fathmm_filtered/filtered_vcf\"\n",
    "aiims_mm_sniper_path = \"/home/vivek/aiims_data_processing/SNIPER/mm_sample_analysis/fathmm_filtered/filtered_vcf\"\n",
    "aiims_mm_varscan_path = \"/home/vivek/aiims_data_processing/varscan2/mm_annovar/vcf_reheader/fathmm/filtered_vcf\"\n",
    "\n",
    "aiims_mgus_muse_path = \"/home/vivek/aiims_data_processing/muse/mgus_sample_analysis/fathmm_filtered/filtered_vcf\"\n",
    "aiims_mgus_mutect_path = \"/home/vivek/aiims_data_processing/MUTECT2/mgus_sample_analysis/fathmm_filtered/filtered_vcf\"\n",
    "aiims_mgus_sniper_path = \"/home/vivek/aiims_data_processing/SNIPER/mgus_sample_analysis/fathmm_filtered/filtered_vcf\"\n",
    "aiims_mgus_varscan_path = \"/home/vivek/aiims_data_processing/varscan2/mgus_annovar/vcf_reheader/fathmm/filtered_vcf\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene798 = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/all_genes_798.txt\").read().split('\\n')\n",
    "gene798 = [i for i in gene798 if i]\n",
    "top500genes = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/ppi_string_merged/new_model4/best_performing_model/top_500_genes.txt\").read().split(\"\\n\")\n",
    "top500genes = [i.split(\"\\t\")[0] for i in top500genes]\n",
    "gene798_rank = pd.read_excel(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/ppi_string_merged/new_model4/best_performing_model/gene798_ranking.xlsx\", index_col=0)\n",
    "gene34 = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/farcastbio/genelist77_mm_mgus.txt\").read().split(\"\\n\")\n",
    "gene34 = [i for i in gene34 if i]\n",
    "gene77 = open(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/farcastbio/77genelist.txt\").read().split(\"\\n\")\n",
    "gene77 = [i for i in gene77 if i]\n",
    "\n",
    "mmrf_sample_list = open('/home/vivek/mmrf_data/mmrf_variant_callers/mmrf_mm_sample_list.txt').read().split('\\n')\n",
    "mmrf_sample_list = [x for x in mmrf_sample_list if not \"PB\" in x]\n",
    "aiims_mm_sample_list = open('/home/vivek/aiims_data_processing/scripts/mm_list.txt').read().split('\\n')\n",
    "aiims_mm_sample_list = [x for x in aiims_mm_sample_list if x != '']\n",
    "aiims_mgus_sample_list = open('/home/vivek/aiims_data_processing/scripts/mgus_list.txt').read().split('\\n')\n",
    "aiims_mgus_sample_list = [x for x in aiims_mgus_sample_list if x != '']\n",
    "ega_sample_list = open('/home/vivek/ega_data_1901/scripts/ega_mgus_list.txt').read().split('\\n')\n",
    "ega_sample_list = [x for x in ega_sample_list if x != '']\n",
    "all_mm = list(set(mmrf_sample_list + aiims_mm_sample_list))\n",
    "all_mm = [x for x in all_mm if x != '']\n",
    "all_mgus = list(set(aiims_mgus_sample_list + ega_sample_list))\n",
    "all_mgus = [x for x in all_mgus if x != '']\n",
    "\n",
    "aiims_mgus_cnv_path = \"/home/vivek/aiims_data_processing/cnvkit_analysis/mgus\"\n",
    "aiims_mm_cnv_path = \"/home/vivek/aiims_data_processing/cnvkit_analysis/mm\"\n",
    "ega_cnv_path = \"/home/vivek/ega_data_1901/cnvkit_analysis\"\n",
    "mmrf_cnv_path = \"/home/vivek/mmrf_data/cnv_analysis/filtered_cns\"\n",
    "\n",
    "with open('/home/vivek/jupyter_notebooks/bio_dgi_extension/cnv_analysis/mm_dict.pkl', 'rb') as f:\n",
    "    mm_cnv_data = pickle.load(f)\n",
    "    \n",
    "with open('/home/vivek/jupyter_notebooks/bio_dgi_extension/cnv_analysis/mgus_dict.pkl', 'rb') as f:\n",
    "    mgus_cnv_data = pickle.load(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_sam_cnvs(sam, disease):\n",
    "    if disease == \"mgus\":\n",
    "        if 'SM' in sam:\n",
    "            df = pd.read_csv(os.path.join(aiims_mgus_cnv_path, sam, sam+\"BM_tumor.call.cns\"), sep=\"\\t\")\n",
    "            df = df[df[\"cn\"]!= 2]\n",
    "            df = df[df[\"gene\"] != '-']\n",
    "            df = df[~df[\"chromosome\"].str.contains(\"random\")]\n",
    "            df = df[~df[\"chromosome\"].str.contains(\"hap\")]\n",
    "            df = df[~df[\"chromosome\"].str.contains(\"chrUn\")].reset_index(drop=True)\n",
    "        elif \"CR\" in sam:\n",
    "            df = pd.read_csv(os.path.join(ega_cnv_path, sam, sam+\"-BM_dedup.realigned.call.cns\"), sep=\"\\t\")\n",
    "            df = df[df[\"cn\"]!= 2]\n",
    "            df = df[df[\"gene\"] != '-']\n",
    "            df = df[~df[\"chromosome\"].str.contains('GL')].reset_index(drop=True)\n",
    "    elif disease == \"mm\":\n",
    "        if 'SM' in sam:\n",
    "            df = pd.read_csv(os.path.join(aiims_mm_cnv_path, sam, sam+\"BM_tumor.call.cns\"), sep=\"\\t\")\n",
    "            df = df[df[\"cn\"]!= 2]\n",
    "            df = df[df[\"gene\"] != '-']\n",
    "            df = df[~df[\"chromosome\"].str.contains(\"random\")]\n",
    "            df = df[~df[\"chromosome\"].str.contains(\"hap\")]\n",
    "            df = df[~df[\"chromosome\"].str.contains(\"chrUn\")].reset_index(drop=True)\n",
    "        if \"MMRF\" in sam:\n",
    "            df = pd.read_csv(os.path.join(mmrf_cnv_path, sam+\"_BM.cns\"), sep=\"\\t\")            \n",
    "            \n",
    "    df['sample_id'] = [sam for i in range(df.shape[0])]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def filter_sam_cnvs(sam_df, gen):\n",
    "    gene_idx = []\n",
    "    for idx in range(sam_df.shape[0]):\n",
    "        try:\n",
    "            string = sam_df.iloc[idx][\"gene\"]\n",
    "            genes = sorted(list(set(ast.literal_eval(string))))\n",
    "        except:\n",
    "            genes = sorted(list(set(sam_df.iloc[idx][\"gene\"].split(','))))\n",
    "        \n",
    "        if gen in genes:\n",
    "            gene_idx.append(idx)\n",
    "    \n",
    "    return sam_df.iloc[gene_idx]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def all_samp_gene_cnv(gene_name, disease):\n",
    "    df_gene_cnv = pd.DataFrame()\n",
    "    if disease == \"mm\":\n",
    "        for sam in all_mm:\n",
    "            try:\n",
    "                # read sample CNVs\n",
    "                df = read_sam_cnvs(sam, disease)\n",
    "                # Filter CNV for a given gene\n",
    "                df = filter_sam_cnvs(df, gene_name)\n",
    "                df_gene_cnv = pd.concat([df_gene_cnv, df]).reset_index(drop=True)\n",
    "            except:\n",
    "                continue\n",
    "    elif disease == \"mgus\":\n",
    "        for sam in all_mgus:\n",
    "            try:\n",
    "                # read sample CNVs\n",
    "                df = read_sam_cnvs(sam, disease)\n",
    "                # Filter CNV for a given gene\n",
    "                df = filter_sam_cnvs(df, gene_name)\n",
    "                df_gene_cnv = pd.concat([df_gene_cnv, df]).reset_index(drop=True)\n",
    "            except:\n",
    "                continue\n",
    "            \n",
    "    return df_gene_cnv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# def all_samp_gene_cnv(gene_name, disease):\n",
    "#     df_gene_cnv = pd.DataFrame()\n",
    "#     if disease == \"mm\":\n",
    "#         for k,v in mm_cnv_data['gene_per_sample_dict'].items():\n",
    "#             if gene_name in v:\n",
    "#                 # read sample CNVs\n",
    "#                 df = read_sam_cnvs(k, disease)\n",
    "#                 # Filter CNV for a given gene\n",
    "#                 df = filter_sam_cnvs(df, gene_name)\n",
    "#                 df_gene_cnv = pd.concat([df_gene_cnv, df]).reset_index(drop=True)\n",
    "#     elif disease == \"mgus\":\n",
    "#         for k,v in mgus_cnv_data['gene_per_sample_dict'].items():\n",
    "#             if gene_name in v:\n",
    "#                 # read sample CNVs\n",
    "#                 df = read_sam_cnvs(k, disease)\n",
    "#                 # Filter CNV for a given gene\n",
    "#                 df = filter_sam_cnvs(df, gene_name)\n",
    "#                 df_gene_cnv = pd.concat([df_gene_cnv, df]).reset_index(drop=True)\n",
    "            \n",
    "#     return df_gene_cnv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_df(sampleid, disease, gene_name):\n",
    "    \n",
    "    if \"MMRF\" in sampleid:\n",
    "        df_muse = pd.read_csv(os.path.join(mmrf_muse_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df_mutect = pd.read_csv(os.path.join(mmrf_mutect_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df_sniper = pd.read_csv(os.path.join(mmrf_sniper_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df_varscan = pd.read_csv(os.path.join(mmrf_varscan_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df = pd.concat([df_muse, df_mutect, df_sniper, df_varscan])\n",
    "        df = df[df[\"Gene_refGene\"] == gene_name]\n",
    "        df = df.drop_duplicates(ignore_index=True)\n",
    "        \n",
    "    if \"CR\" in sampleid:\n",
    "        df_muse = pd.read_csv(os.path.join(ega_muse_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df_mutect = pd.read_csv(os.path.join(ega_mutect_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df_sniper = pd.read_csv(os.path.join(ega_sniper_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df_varscan = pd.read_csv(os.path.join(ega_varscan_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "        df = pd.concat([df_muse, df_mutect, df_sniper, df_varscan])\n",
    "        df = df[df[\"Gene_refGene\"] == gene_name]\n",
    "        df = df.drop_duplicates()    \n",
    "    \n",
    "    if \"SM\" in sampleid:\n",
    "        if disease == \"mm\":\n",
    "            df_muse = pd.read_csv(os.path.join(aiims_mm_muse_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df_mutect = pd.read_csv(os.path.join(aiims_mm_mutect_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df_sniper = pd.read_csv(os.path.join(aiims_mm_sniper_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df_varscan = pd.read_csv(os.path.join(aiims_mm_varscan_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df = pd.concat([df_muse, df_mutect, df_sniper, df_varscan])\n",
    "            df = df[df[\"Gene_refGene\"] == gene_name]\n",
    "            df = df.drop_duplicates()\n",
    "        elif disease == \"mgus\":\n",
    "            df_muse = pd.read_csv(os.path.join(aiims_mgus_muse_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df_mutect = pd.read_csv(os.path.join(aiims_mgus_mutect_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df_sniper = pd.read_csv(os.path.join(aiims_mgus_sniper_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df_varscan = pd.read_csv(os.path.join(aiims_mgus_varscan_path, sampleid + \".vcf\"), sep=\"\\t\")\n",
    "            df = pd.concat([df_muse, df_mutect, df_sniper, df_varscan])\n",
    "            df = df[df[\"Gene_refGene\"] == gene_name]\n",
    "            df = df.drop_duplicates()\n",
    "        \n",
    "    df['sample_id'] = [sampleid for i in range(len(df))]\n",
    "    \n",
    "    return df\n",
    "        \n",
    "        \n",
    "def get_gene_df_all_samples(gene_name, disease, save_out = False):\n",
    "    \n",
    "    df_gene = pd.DataFrame()\n",
    "    df_gene1 = pd.DataFrame()\n",
    "    mm_sample_count, mgus_sample_count = 0,0\n",
    "    if disease == \"mm\":\n",
    "        for sam in all_mm:\n",
    "            df = get_gene_df(sam, \"mm\", gene_name)\n",
    "            if df.shape[0] > 0:\n",
    "                mm_sample_count += 1\n",
    "                df_gene = pd.concat([df_gene, df])\n",
    "\n",
    "        if df_gene.shape[0] > 0:\n",
    "            df_gene[\"chr\"] = [i.split(\"chr\")[-1] for i in df_gene[\"chr\"]]\n",
    "            df_gene = df_gene.drop_duplicates(ignore_index=True)\n",
    "            df_gene = df_gene[[\"sample_id\",\"chr\",\"pos\",\"ref\",\"alt\",\"Gene_refGene\",\"Func_refGene\",\"Exonic_refGene\",\"AF\",\"AD\"]]\n",
    "            df_gene1 = df_gene\n",
    "            # df_gene1 = df_gene.groupby(['pos']).agg(lambda x: list(x))\n",
    "        \n",
    "            if save_out:\n",
    "                df_gene1.to_excel(os.path.join(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene_snvs/ungrouped_sample_level/mm\", gene_name + \".xlsx\"), index=False)\n",
    "        \n",
    "        return df_gene1, mm_sample_count\n",
    "        \n",
    "    elif disease == \"mgus\":\n",
    "        for sam in all_mgus:\n",
    "            df = get_gene_df(sam, \"mgus\", gene_name)\n",
    "            if df.shape[0] > 0:\n",
    "                mgus_sample_count += 1\n",
    "                df_gene = pd.concat([df_gene, df])\n",
    "\n",
    "        if df_gene.shape[0] > 0:\n",
    "            df_gene[\"chr\"] = [i.split(\"chr\")[-1] for i in df_gene[\"chr\"]]\n",
    "            df_gene = df_gene.drop_duplicates(ignore_index=True)\n",
    "            df_gene = df_gene[[\"sample_id\",\"chr\",\"pos\",\"ref\",\"alt\",\"Gene_refGene\",\"Func_refGene\",\"Exonic_refGene\",\"AF\",\"AD\"]]\n",
    "            df_gene1 = df_gene\n",
    "            # df_gene1 = df_gene.groupby(['pos']).agg(lambda x: list(x))\n",
    "    \n",
    "            if save_out:\n",
    "                file_path = os.path.join(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene_snvs/\", disease, gene_name+\".xlsx\")\n",
    "                if not os.path.exists(file_path):\n",
    "                    df_gene1.to_excel(file_path, index=False)\n",
    "            \n",
    "            # if save_out:\n",
    "            #     df_gene1.to_excel(os.path.join(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene_snvs/ungrouped_sample_level/mgus\", gene_name + \".xlsx\"), index=False)\n",
    "            \n",
    "        return df_gene1, mgus_sample_count\n",
    "    \n",
    "\n",
    "def get_gene_df_filtered(df_gene1, disease, gname, save_out = True):\n",
    "    df_gene2 = pd.DataFrame()\n",
    "    chr2, ref2,alt2, gene2, func2, exonic2, af2, ad2 = [],[],[],[],[],[],[],[]\n",
    "    for idx in range(df_gene1.shape[0]):\n",
    "        \n",
    "        chr2.append(df_gene1.iloc[idx][\"chr\"][0])\n",
    "        \n",
    "        if isinstance(df_gene1.iloc[idx][\"ref\"], list):\n",
    "            if list(set(df_gene1.iloc[idx][\"ref\"])).__len__() > 1:\n",
    "                ref2.append('/'.join(list(set(df_gene1.iloc[idx][\"ref\"]))))\n",
    "            else:\n",
    "                ref2.append(df_gene1.iloc[idx][\"ref\"][0])\n",
    "        else:\n",
    "            ref2.append(df_gene1.iloc[idx][\"ref\"])\n",
    "        \n",
    "        if isinstance(df_gene1.iloc[idx][\"alt\"], list):\n",
    "            if list(set(df_gene1.iloc[idx][\"alt\"])).__len__() > 1:\n",
    "                alt2.append('/'.join(list(set(df_gene1.iloc[idx][\"alt\"]))))\n",
    "            else:\n",
    "                alt2.append(df_gene1.iloc[idx][\"alt\"][0])\n",
    "        else:\n",
    "            alt2.append(df_gene1.iloc[idx][\"alt\"])\n",
    "        \n",
    "        if isinstance(df_gene1.iloc[idx][\"Gene_refGene\"], list):\n",
    "            gene2.append(df_gene1.iloc[idx][\"Gene_refGene\"][0])\n",
    "        else:\n",
    "            gene2.append(df_gene1.iloc[idx][\"Gene_refGene\"])\n",
    "        \n",
    "        if isinstance(df_gene1.iloc[idx][\"Func_refGene\"], list):\n",
    "            if list(set(df_gene1.iloc[idx][\"Func_refGene\"])).__len__() > 1:\n",
    "                func2.append('/'.join(list(set(df_gene1.iloc[idx][\"Func_refGene\"]))))\n",
    "            else:\n",
    "                func2.append(df_gene1.iloc[idx][\"Func_refGene\"][0])\n",
    "        else:\n",
    "            func2.append(df_gene1.iloc[idx][\"Func_refGene\"])\n",
    "            \n",
    "        if isinstance(df_gene1.iloc[idx][\"Exonic_refGene\"], list):\n",
    "            if list(set(df_gene1.iloc[idx][\"Exonic_refGene\"])).__len__() > 1:\n",
    "                    exonic2.append('/'.join(list(set(df_gene1.iloc[idx][\"Exonic_refGene\"]))))\n",
    "            else:\n",
    "                exonic2.append(df_gene1.iloc[idx][\"Exonic_refGene\"][0])\n",
    "        else:\n",
    "            exonic2.append(df_gene1.iloc[idx][\"Exonic_refGene\"])\n",
    "            \n",
    "        \n",
    "        if isinstance(df_gene1.iloc[idx][\"AF\"], list):\n",
    "            if df_gene1.iloc[idx][\"AF\"].__len__() > 1:\n",
    "                print(\"this\")\n",
    "                # print(df_gene1.iloc[idx][\"AF\"])\n",
    "                aaf = df_gene1.iloc[idx][\"AF\"]\n",
    "                af2.append(np.median([float(aaf[i].split(',')[0]) if isinstance(aaf[i],str) else float(aaf[i]) for i in range(aaf.__len__())]))\n",
    "            else:\n",
    "                print(\"this0\")\n",
    "                if ',' in df_gene1.iloc[idx][\"AF\"]:\n",
    "                    afs = [float(i) for i in df_gene1.iloc[idx][\"AF\"].split(',')]\n",
    "                    af2.append(np.median(afs))\n",
    "                else:\n",
    "                    print(\"this00\")\n",
    "                    af2.append(df_gene1.iloc[idx][\"AF\"][0])\n",
    "        else:\n",
    "            af2.append(df_gene1.iloc[idx][\"AF\"])\n",
    "            \n",
    "        if isinstance(df_gene1.iloc[idx][\"AD\"], list):\n",
    "            if df_gene1.iloc[idx][\"AD\"].__len__() > 1:\n",
    "                aad = df_gene1.iloc[idx][\"AD\"]\n",
    "                ad2.append(np.median([float(aad[i].split(',')[0]) if isinstance(aad[i],str) else float(aad[i]) for i in range(aad.__len__())]))\n",
    "                # ad2.append(np.median([float(i) for i in df_gene1.iloc[idx][\"AD\"]]))\n",
    "            else:\n",
    "                if ',' in df_gene1.iloc[idx][\"AD\"]:\n",
    "                    ads = [float(i) for i in df_gene1.iloc[idx][\"AD\"].split()]\n",
    "                    ad2.append(np.median(ads))\n",
    "                else:\n",
    "                    ad2.append(df_gene1.iloc[idx][\"AD\"][0])\n",
    "        else:\n",
    "            ad2.append(df_gene1.iloc[idx][\"AD\"])\n",
    "            \n",
    "    df_gene2[\"chr\"] = chr2\n",
    "    df_gene2[\"pos\"] = df_gene1.index.tolist()\n",
    "    df_gene2[\"ref\"] = ref2\n",
    "    df_gene2[\"alt\"] = alt2\n",
    "    df_gene2[\"gene\"] = gene2\n",
    "    df_gene2[\"func\"] = func2\n",
    "    df_gene2[\"exonic\"] = exonic2\n",
    "    df_gene2[\"af\"] = af2\n",
    "    df_gene2[\"ad\"] = ad2\n",
    "    \n",
    "    if save_out:\n",
    "        file_path = os.path.join(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene_snvs/\", disease, gname+\".xlsx\")\n",
    "        if not os.path.exists(file_path):\n",
    "            df_gene2.to_excel(file_path, index=False)\n",
    "    return df_gene2\n",
    "\n",
    "def get_gene_processed_snvs(gene_name, disease):\n",
    "    file_path = os.path.join(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/gene_snvs/\", disease, gene_name+\".xlsx\")\n",
    "    if os.path.exists(file_path):\n",
    "        df = pd.read_excel(file_path)\n",
    "    else:\n",
    "        df_mm, _ = get_gene_df_all_samples(gene_name, disease, save_out=False)\n",
    "        df = get_gene_df_filtered(df_mm, disease, gene_name, save_out=True)\n",
    "    return df\n",
    "\n",
    "def filter_gene_snvs_in_cnvs(df_snvs, start_pos, end_pos):\n",
    "    df_snvs = df_snvs.query('pos >='+ str(start_pos) +'  & pos <= ' + str(end_pos))\n",
    "    return df_snvs    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df = get_gene_processed_snvs(\"MYC\", \"mm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "chess_data = pd.read_csv('/home/vivek/jupyter_notebooks/bio_dgi_extension/farcastbio/chess_database/chess3.0.1.gff', sep='\\t', header=None)\n",
    "chess_data.columns = ['chr', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase', 'attributes']\n",
    "chess_data['gene_id'] = chess_data['attributes'].apply(lambda x: x.split('gene_name')[-1].split(';')[0].split('=')[-1])\n",
    "chess_data['id'] = chess_data['attributes'].apply(lambda x: x.split('ID')[-1].split(';')[0].split('=')[-1])\n",
    "chess_data = chess_data.drop([\"score\"], axis=1)\n",
    "# genelist = chess_data[\"gene_id\"].unique().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_cds_deletion(chess_gene, start, end):\n",
    "    # total_transcripts = chess_gene[chess_gene[\"type\"] == \"transcript\"].shape[0]\n",
    "    gene_unique_id = chess_gene[\"id\"].unique().tolist()\n",
    "    del_true = 0\n",
    "    for uid in gene_unique_id:\n",
    "        chess_gene_uid_cds = chess_gene[(chess_gene[\"id\"] == uid) & (chess_gene[\"type\"] == \"CDS\")]\n",
    "        total_cds_len = sum([j-i for i,j in zip(chess_gene_uid_cds[\"start\"].tolist(), chess_gene_uid_cds[\"end\"].tolist())])\n",
    "        deL_cds_len = sum([j-i for i,j in zip(chess_gene_uid_cds[\"start\"].tolist(), chess_gene_uid_cds[\"end\"].tolist()) if (i >= start) & (j <= end)])\n",
    "        if deL_cds_len > total_cds_len/2:\n",
    "            del_true += 1\n",
    "        # else:\n",
    "        #     del_false += 1\n",
    "            \n",
    "    if del_true == gene_unique_id.__len__():\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_first_exon_deletion(chess_gene, start, end):\n",
    "    gene_unique_id = chess_gene[\"id\"].unique().tolist()\n",
    "    del_true = 0\n",
    "    for uid in gene_unique_id:\n",
    "        chess_gene_uid_exon = chess_gene[(chess_gene[\"id\"] == uid) & (chess_gene[\"type\"] == \"exon\")].sort_values(by=['start'])\n",
    "        chess_gene_uid_first_exon_start = chess_gene_uid_exon[\"start\"].tolist()[0]\n",
    "        chess_gene_uid_first_exon_end = chess_gene_uid_exon[\"end\"].tolist()[0]\n",
    "        if (chess_gene_uid_first_exon_start >= start) & (chess_gene_uid_first_exon_end <= end):\n",
    "            del_true += 1\n",
    "        # else:\n",
    "        #     del_false += 1\n",
    "            \n",
    "    if del_true == gene_unique_id.__len__():\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_splice_variants(chess_gene, start, end, snv_df):\n",
    "    gene_unique_id = chess_gene[\"id\"].unique().tolist()\n",
    "    del_true = 0\n",
    "    for uid in gene_unique_id:\n",
    "        chess_gene_uid_exon = chess_gene[(chess_gene[\"id\"] == uid) & (chess_gene[\"type\"] == \"transcript\")].sort_values(by=['start'])\n",
    "        chess_gene_uid_start = chess_gene_uid_exon[\"start\"].tolist()[0]\n",
    "        chess_gene_uid_end = chess_gene_uid_exon[\"end\"].tolist()[0]\n",
    "        snv_filt_df = filter_gene_snvs_in_cnvs(snv_df, chess_gene_uid_start, chess_gene_uid_start)\n",
    "        if snv_filt_df.shape[0] > 0:\n",
    "            snv_filt_df = snv_filt_df[snv_filt_df[\"func\"].str.contians(\"splice\")]\n",
    "            if snv_filt_df.shape[0] > 0:\n",
    "                del_true += 1\n",
    "\n",
    "    if del_true == gene_unique_id.__len__():\n",
    "        return \"True\"\n",
    "    else:\n",
    "        return \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_frameshift_variants(chess_gene, start, end, snv_df):\n",
    "    gene_unique_id = chess_gene[\"id\"].unique().tolist()\n",
    "    del_true = 0\n",
    "    for uid in gene_unique_id:\n",
    "        chess_gene_uid_exon = chess_gene[(chess_gene[\"id\"] == uid) & (chess_gene[\"type\"] == \"transcript\")].sort_values(by=['start'])\n",
    "        chess_gene_uid_start = chess_gene_uid_exon[\"start\"].tolist()[0]\n",
    "        chess_gene_uid_end = chess_gene_uid_exon[\"end\"].tolist()[0]\n",
    "        snv_filt_df = filter_gene_snvs_in_cnvs(snv_df, chess_gene_uid_start, chess_gene_uid_start)\n",
    "        if snv_filt_df.shape[0] > 0:\n",
    "            snv_filt_df = snv_filt_df[snv_filt_df[\"func\"].str.contians(\"frameshift\")]\n",
    "            if snv_filt_df.shape[0] > 0:\n",
    "                del_true += 1\n",
    "\n",
    "    if del_true == gene_unique_id.__len__():\n",
    "        return \"True\"\n",
    "    else:\n",
    "        return \"False\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def lof_pipeline(gene_name, disease):\n",
    "    lof_sample_list = []\n",
    "    df_cnv = all_samp_gene_cnv(gene_name, disease)\n",
    "    if df_cnv.shape[0] > 0:\n",
    "        df_cnv_del = df_cnv[df_cnv[\"cn\"] < 2]\n",
    "        chess_gene = chess_data[chess_data[\"gene_id\"] == gene_name]\n",
    "        snv_df = get_gene_processed_snvs(gene_name, disease)\n",
    "        if df_cnv_del.shape[0] > 0:\n",
    "            for idx in range(df_cnv_del.shape[0]):\n",
    "                sam_id = df_cnv_del.iloc[idx][\"sample_id\"]\n",
    "                cnv_start = df_cnv_del.iloc[idx][\"start\"]\n",
    "                cnv_end = df_cnv_del.iloc[idx][\"end\"]\n",
    "                case_a = find_cds_deletion(chess_gene, cnv_start, cnv_end)\n",
    "                case_b = find_first_exon_deletion(chess_gene, cnv_start, cnv_end)\n",
    "                case_c = find_splice_variants(chess_gene, cnv_start, cnv_end, snv_df)\n",
    "                case_d = find_frameshift_variants(chess_gene, cnv_start, cnv_end, snv_df)\n",
    "                if case_a or case_b or case_c or case_d:\n",
    "                    lof_sample_list.append(sam_id)\n",
    "            \n",
    "            \n",
    "    lof_sample_list = list(set(lof_sample_list))\n",
    "    return lof_sample_list.__len__(), lof_sample_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# lof_pipeline(\"MYC\", \"mm\")\n",
    "# gene625_lof = pd.read_excel(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/lof/lof_gene625.xlsx\", index_col=0)\n",
    "# gene625_lof.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gene625_lof.loc[\"ABCA3\",\"lof_mm\"]\n",
    "# lof_pipeline(\"ADH6\", \"mm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 798/798 [1:11:57<00:00,  5.41s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>lof_mm</th>\n",
       "      <th>lof_mgus</th>\n",
       "      <th>lof_mm_sample_list</th>\n",
       "      <th>lof_mgus_sample_list</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCA3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>['SM0029', 'SM0172', 'SM0018']</td>\n",
       "      <td>['CR-MGUS-88_88', 'CR-MGUS-70_70', 'CR-MGUS-19...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>['SM0029', 'SM0172', 'SM0018']</td>\n",
       "      <td>['CR-MGUS-71_71', 'SM1490', 'SM1731']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_name  lof_mm  lof_mgus              lof_mm_sample_list  \\\n",
       "0     ABCA1       0         0                              []   \n",
       "1     ABCA3       3         9  ['SM0029', 'SM0172', 'SM0018']   \n",
       "2     ABCA7       3         3  ['SM0029', 'SM0172', 'SM0018']   \n",
       "3      ABL2       0         0                              []   \n",
       "4     ACACB       0         0                              []   \n",
       "\n",
       "                                lof_mgus_sample_list  \n",
       "0                                                 []  \n",
       "1  ['CR-MGUS-88_88', 'CR-MGUS-70_70', 'CR-MGUS-19...  \n",
       "2              ['CR-MGUS-71_71', 'SM1490', 'SM1731']  \n",
       "3                                                 []  \n",
       "4                                                 []  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lof798 = pd.DataFrame()\n",
    "gene701_lof = pd.read_excel(\"/home/vivek/jupyter_notebooks/bio_dgi_extension/final_compilation_work/lof/lof_gene701.xlsx\", index_col=0)\n",
    "lof_mm, lof_mgus = [], []\n",
    "lof_mm_sample_list, lof_mgus_sample_list = [], []\n",
    "genelist = gene798\n",
    "for gen in tqdm(genelist):\n",
    "    if gen in gene701_lof.index.tolist():\n",
    "        lof_mm.append(gene701_lof.loc[gen,\"lof_mm\"])\n",
    "        lof_mgus.append(gene701_lof.loc[gen,\"lof_mgus\"])\n",
    "        lof_mm_sample_list.append(gene701_lof.loc[gen,\"lof_mm_sample_list\"])\n",
    "        lof_mgus_sample_list.append(gene701_lof.loc[gen,\"lof_mgus_sample_list\"])\n",
    "    else:\n",
    "        try:\n",
    "            \n",
    "            no_mm_sample, lof_mm_sam = lof_pipeline(gen, \"mm\")\n",
    "            lof_mm.append(no_mm_sample)\n",
    "            lof_mm_sample_list.append(lof_mm_sam)\n",
    "        except:\n",
    "            lof_mm.append(0)\n",
    "            lof_mm_sample_list.append([])\n",
    "        try:\n",
    "            no_mgus_sample, lof_mgus_sam = lof_pipeline(gen, \"mgus\")\n",
    "            lof_mgus.append(no_mgus_sample)\n",
    "            lof_mgus_sample_list.append(lof_mgus_sam)\n",
    "        except:\n",
    "            lof_mgus.append(0)\n",
    "            lof_mgus_sample_list.append([])\n",
    "    \n",
    "df_lof798[\"gene_name\"] = genelist\n",
    "df_lof798[\"lof_mm\"] = lof_mm\n",
    "df_lof798[\"lof_mgus\"] = lof_mgus\n",
    "df_lof798[\"lof_mm_sample_list\"] = lof_mm_sample_list\n",
    "df_lof798[\"lof_mgus_sample_list\"] = lof_mgus_sample_list\n",
    "# df_lof500 = df_lof500[(df_lof500[\"lof_mm\"] > 0) & (df_lof500[\"lof_mgus\"] > 0)]\n",
    "df_lof798.to_excel(\"lof798.xlsx\", index=False)\n",
    "df_lof798.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_lof798 = pd.read_excel(\"../lof/lof798.xlsx\")\n",
    "ghis_scores =pd.read_excel(\"../haploinsufficiency/GHIS_scores.xlsx\")\n",
    "df_lof798[\"ghis_score\"] = [ghis_scores[ghis_scores[\"name\"] == gene][\"GHIS\"].values[0]  if gene in ghis_scores[\"name\"].values.tolist() else np.nan for gene in df_lof798[\"gene_name\"]]\n",
    "df_lof798.to_excel(\"lof798_ghis.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_name</th>\n",
       "      <th>lof_mm</th>\n",
       "      <th>lof_mgus</th>\n",
       "      <th>lof_mm_sample_list</th>\n",
       "      <th>lof_mgus_sample_list</th>\n",
       "      <th>ghis_score</th>\n",
       "      <th>rank</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.459994</td>\n",
       "      <td>65</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCA3</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>['SM0029', 'SM0172', 'SM0018']</td>\n",
       "      <td>['CR-MGUS-88_88', 'CR-MGUS-70_70', 'CR-MGUS-19...</td>\n",
       "      <td>0.488285</td>\n",
       "      <td>332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABCA7</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>['SM0029', 'SM0172', 'SM0018']</td>\n",
       "      <td>['CR-MGUS-71_71', 'SM1490', 'SM1731']</td>\n",
       "      <td>0.520105</td>\n",
       "      <td>441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.533039</td>\n",
       "      <td>175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACACB</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.565962</td>\n",
       "      <td>339</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene_name  lof_mm  lof_mgus              lof_mm_sample_list  \\\n",
       "0     ABCA1       0         0                              []   \n",
       "1     ABCA3       3         9  ['SM0029', 'SM0172', 'SM0018']   \n",
       "2     ABCA7       3         3  ['SM0029', 'SM0172', 'SM0018']   \n",
       "3      ABL2       0         0                              []   \n",
       "4     ACACB       0         0                              []   \n",
       "\n",
       "                                lof_mgus_sample_list  ghis_score  rank  \n",
       "0                                                 []    0.459994    65  \n",
       "1  ['CR-MGUS-88_88', 'CR-MGUS-70_70', 'CR-MGUS-19...    0.488285   332  \n",
       "2              ['CR-MGUS-71_71', 'SM1490', 'SM1731']    0.520105   441  \n",
       "3                                                 []    0.533039   175  \n",
       "4                                                 []    0.565962   339  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_lof798 = pd.read_excel(\"../lof/lof798_ghis.xlsx\")\n",
    "df_lof798[\"rank\"] = [gene798_rank.loc[gene,\"rank\"] if gene in gene798_rank.index.tolist() else np.nan for gene in df_lof798[\"gene_name\"]]\n",
    "df_lof798.to_excel(\"lof798_ghis.xlsx\", index=False)\n",
    "df_lof798.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
